test/schedulers/time_scheduler.cpp

```diff
@@ -265,3 +272,72 @@ TEST_CASE("time_scheduler produces set_stopped debug signal",
     CHECK(debug_events ==
           std::vector{"op sched start"s, "op sched set_stopped"s});
 }
+
+namespace {
+template <typename Domain> struct std_hal {
```

Comment:
question: do we want to keep this here or make it part of the library?

Reply:
The one here is a proof of concept but I don't know that it is robust enough to be part of the library?